azure/durable_functions/models/DurableOrchestrationStatus.py

```diff
@@ -13,20 +12,23 @@ class DurableOrchestrationStatus:
     """
 
     # parameter names are as defined by JSON schema and do not conform to PEP8 naming conventions
-    def __init__(self, name: str = None, instanceId: str = None, createdTime: str = None,
-                 lastUpdatedTime: str = None, input: Any = None, output: Any = None,
-                 runtimeStatus: str = None, customStatus: Any = None, history: List[Any] = None,
+    def __init__(self, name: Optional[str] = None, instanceId: Optional[str] = None,
+                 createdTime: Optional[str] = None, lastUpdatedTime: Optional[str] = None,
+                 input: Optional[Any] = None, output: Optional[Any] = None,
+                 runtimeStatus: Optional[str] = None, customStatus: Optional[Any] = None,
+                 history: Optional[List[Any]] = None,
                  **kwargs):
-        self._name: str = name
-        self._instance_id: str = instanceId
-        self._created_time: datetime = dt_parse(createdTime) if createdTime is not None else None
-        self._last_updated_time: datetime = dt_parse(lastUpdatedTime) \
+        self._name: Optional[str] = name
+        self._instance_id: Optional[str] = instanceId
+        self._created_time: Optional[datetime] = \
+            dt_parse(createdTime) if createdTime is not None else None
+        self._last_updated_time: Optional[datetime] = dt_parse(lastUpdatedTime) \
             if lastUpdatedTime is not None else None
         self._input: Any = input
         self._output: Any = output
-        self._runtime_status: OrchestrationRuntimeStatus = runtimeStatus
+        self._runtime_status: Optional[str] = runtimeStatus
```

Comment:
why the switch to string here?

Reply:
As it turns out, the runtimeStatus passed in was never of type `OrchestrationRuntimeStatus`, it was always a string. It becomes an `OrchestrationRuntimeStatus` datatype *later* in the execution. In particular, it happens here: https://github.com/Azure/azure-functions-durable-python/blob/dev/azure/durable_functions/models/DurableOrchestrationClient.py#L433

**I want to fix this in the future**, but I remember trying to eagerly construct an `OrchestrationRuntimeStatus` out of this variable for this PR and that was causing problems down the line, so I don't think it's worth fixing just yet.

Reply:
As long as we fix it before GA and we are stuck with supporting it, that is fine to me. Maybe leave a TODO with the issue number for this?

Reply:
Done, with a reference to: #178